Xunit migration

galvesribeiro opened the pull request:
Just migrated all tests from MSTest to Xunit as metioned on #9.

Note that now all test content is on test/ directory and source code on source/ directory. This move is necessary in order to run xunit on .Net Core later on.

Appreciate feedback.

Thanks

nehmebilal commented:
Took a quick look. Looks pretty good! Few things I noticed:

* It's a bit weird to have the .sln file outside the src folder. Can we have both Etg.Yams and Etg.Yams.Test inside the src folder?
* The exes are not actually checked-in. They get compiled when the Etg.Yams.Test is compiled.
* Thanks for cleaning up the asserts on exceptions :)
* I noticed that in some cases you used a text fixture and others you didn't (I think I saw at least one test class that didn't use a fixture). Are the fixtures necessary when they are not shared?
* On the note above, I noticed that most of the tests that starts the blob storage emulator are very similar. Can we reuse the same fixture? Fixtures that need to add additional logic can inherit from that StorageEmulatorTestFixture. Seems like test fixtures make sharing that code very convenient.

Thanks again @galvesribeiro!! Will take a closer look tomorrow.

galvesribeiro commented:
> It's a bit weird to have the .sln file outside the src folder. Can we have both Etg.Yams and Etg.Yams.Test inside the src folder?

This is very common on other projects. The solution has source projects which produces the artifacts and test projects. The xunit runner discover tests on .Net Core based on that. We did it in Orleans as well.

> The exes are not actually checked-in. They get compiled when the Etg.Yams.Test is compiled.

AHHH ok. Sorry will fix that.

> Thanks for cleaning up the asserts on exceptions 

Np. I think since there is an small number of tests, I should convert the Asserts from MSTest to xunit assertion APIs. Orleans has a ton (1000+) tests so for the first migration we changed the runner for xunit but kept the Assertion from MSTest so the merge would be made in two steps however, if we keep MStest assertion APIs, the error messages on test results are not so helpful and thats why I moved it.

> I noticed that in some cases you used a text fixture and others you didn't (I think I saw at least one test class that didn't use a fixture). Are the fixtures necessary when they are not shared?

If you have a look on the tests classes I'm using `IClassFixture<TFixture>` which means that the fixture class will be created once and shared across all tests. It is the xunit relative for `[ClassInitializer]`.  The classes which don't had fixture has a small number of immutable strings so I don't think there is a problem on keep them on the ctor (which is executed on every test method).

> On the note above, I noticed that most of the tests that starts the blob storage emulator are very similar. Can we reuse the same fixture? Fixtures that need to add additional logic can inherit from that StorageEmulatorTestFixture. Seems like test fixtures make sharing that code very convenient.

Yes. I did not changed the way the tests are being made, just blindly ported the MSTest -> xunit in terms of API calls. I agree that if we can share the same StorageEmulator instance, we could have base class which initializes it and also if multiple tests can run on that same run, we can put all of the in a single collection. I just need to understand with you want can run with a shared StorageEmulator context and which one needs a clean storage so we can create a fixture for each one and even let them run in parallel.

Note that since I saw the storage being cleaned on several different tests, I had disabled the parallel tests and not shared anything however, you can help me on explain/list the tests which can share the context, and the ones that don't so I can arrange them on their own fixtures shared or not inside a `ICollectionFixture<TFixture>`.

I hope it help! :)

nehmebilal commented on test/AzureTestUtils/AzureTestUtils.csproj:
```diff
@@ -1,51 +1,48 @@
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>..\Etg.Yams.Test\Data\Exes\</OutputPath>
```
This is why you had issues with the exes.

nehmebilal commented on test/AzureTestUtils/AzureTestUtils.csproj:
```diff
@@ -1,51 +1,48 @@
-    <PlatformTarget>AnyCPU</PlatformTarget>
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>..\Etg.Yams.Test\Data\Exes\</OutputPath>
```
idem

nehmebilal commented on test/HangingProcess/HangingProcess.csproj:
```diff
@@ -1,23 +1,24 @@
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>..\Etg.Yams.Test\Data\Exes\</OutputPath>
```
Restore

nehmebilal commented on test/SuicidalProcess/SuicidalProcess.csproj:
```diff
@@ -1,23 +1,24 @@
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>..\Etg.Yams.Test\Data\Exes\</OutputPath>
```
Restore

nehmebilal commented on test/SuicidalProcess/SuicidalProcess.csproj:
```diff
@@ -26,7 +27,7 @@
     <PlatformTarget>AnyCPU</PlatformTarget>
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>..\Etg.Yams.Test\Data\Exes\</OutputPath>
```
Restore

nehmebilal commented on test/TestProcess/TestProcess.csproj:
```diff
@@ -1,23 +1,24 @@
     <DebugSymbols>true</DebugSymbols>
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
-    <OutputPath>..\Etg.Yams.Test\Data\Exes\</OutputPath>
```
Restore

nehmebilal commented on test/TestProcess/TestProcess.csproj:
```diff
@@ -26,7 +27,7 @@
     <PlatformTarget>AnyCPU</PlatformTarget>
     <DebugType>pdbonly</DebugType>
     <Optimize>true</Optimize>
-    <OutputPath>..\Etg.Yams.Test\Data\Exes\</OutputPath>
```
Restore

nehmebilal commented:
Looks great, thanks @galvesribeiro!
